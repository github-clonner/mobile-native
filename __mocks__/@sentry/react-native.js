export const captureMessage = jest.fn();
export default {};
