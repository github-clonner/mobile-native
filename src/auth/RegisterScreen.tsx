//@ts-nocheck
import React, { Component } from 'react';

import { View } from 'react-native';

import RegisterForm from './RegisterForm';
import logService from '../common/services/log.service';
import ThemedStyles from '../styles/ThemedStyles';

/**
 * Register screen
 */
export default class RegisterScreen extends Component {
  /**
   * Disable navigation bar
   */
  static navigationOptions = {
    header: null,
  };

  /**
   * Render
   */
  render() {
    return (
      <View style={[ThemedStyles.style.flexContainer]}>
        <RegisterForm onRegister={this.onRegister} onBack={this.onPressBack} />
      </View>
    );
  }

  /**
   * On press back
   */
  onPressBack = () => {
    this.props.navigation.navigate('Login');
  };

  /**
   * On register
   */
  onRegister = (guid) => {
    logService.info('[Register] new user registered ' + guid);
  };
}
