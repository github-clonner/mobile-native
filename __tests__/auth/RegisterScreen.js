import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import RegisterScreen from '../../src/auth/RegisterScreen';

jest.mock('../../src/auth/RegisterForm', () => 'RegisterForm');

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('RegisterScreen component', () => {
  it('should renders correctly', () => {
    const loginScreen = renderer.create(
      <RegisterScreen />
    ).toJSON();
    expect(loginScreen).toMatchSnapshot();
  });

  it('should shows register form component', async () => {
    const wrapper = shallow(
      <RegisterScreen />
    );

    // find register form
    const registerForms = wrapper.find('RegisterForm');

    // should contain 1 login form
    expect(registerForms.length).toBe(1);
  });

  it('should navigate on RegisterFormNew events', async () => {

    const navigation = {
      dispatch: jest.fn(),
      navigate: jest.fn(),
    };

    const wrapper = shallow(
      <RegisterScreen navigation={navigation}/>
    );

    // find register form
    const registerForms = wrapper.find('RegisterForm');

    // should contain 1 login form
    expect(registerForms.length).toBe(1);

    // simulate onBack event
    registerForms.at(0).props().onBack();

    // with a navigation action with route to Login
    expect(navigation.navigate).toBeCalledWith('Login');
  });
});
