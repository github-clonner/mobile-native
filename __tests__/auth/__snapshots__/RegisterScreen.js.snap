// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterScreen component should renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <RegisterForm
    onBack={[Function]}
    onRegister={[Function]}
  />
</View>
`;
